src/renderer/components/+add-cluster/add-cluster.tsx

```diff
@@ -128,33 +130,49 @@ export class AddCluster extends React.Component {
       const newClusters: ClusterModel[] = this.selectedContexts.map(context => {
         const clusterId = uuid();
         const kubeConfig = this.kubeContexts.get(context);
-        const kubeConfigPath = this.sourceTab === KubeConfigSourceTab.FILE
-          ? this.kubeConfigPath // save link to original kubeconfig in file-system
-          : ClusterStore.embedCustomKubeConfig(clusterId, kubeConfig); // save in app-files folder
-        return {
-          id: clusterId,
-          kubeConfigPath: kubeConfigPath,
-          workspace: workspaceStore.currentWorkspaceId,
-          contextName: kubeConfig.currentContext,
-          preferences: {
-            clusterName: kubeConfig.currentContext,
-            httpsProxy: this.proxyServer || undefined,
-          },
+        const validateConfig = validateKubeConfig(kubeConfig);
+        if (validateConfig === "OK") { 
+          const kubeConfigPath = this.sourceTab === KubeConfigSourceTab.FILE
+            ? this.kubeConfigPath // save link to original kubeconfig in file-system
+            : ClusterStore.embedCustomKubeConfig(clusterId, kubeConfig); // save in app-files folder
+          return {
+            id: clusterId,
+            kubeConfigPath: kubeConfigPath,
+            workspace: workspaceStore.currentWorkspaceId,
+            contextName: kubeConfig.currentContext,
+            preferences: {
+              clusterName: kubeConfig.currentContext,
+              httpsProxy: this.proxyServer || undefined,
+            },
+          }
+        } else {
+          configValidationErrors.push(
+            `Kubeconfig Context ${JSON.stringify(kubeConfig.contexts[0].name)} - failed validation: "${validateConfig}"\n\n`
+          );
         }
+      }).filter(function (cluster) {
+        return typeof cluster !== "undefined"
```

Comment:
I think this could be shortened

```
.filter(cluster => !!cluster)
```

Reply:
Actually refactored and moved the validation of the kube config into the filter. This makes it a little cleaner as only configs that pass validation will be processed.